iwyu.cc

```diff
@@ -1699,6 +1700,17 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
       ReportTypeUse(used_loc, type);
   }
 
+  void ReportNamespaceAliasUse(SourceLocation used_loc,
```

Comment:
Can we skip the `Track unused namespace aliases` commit for now? I'm not sure if it's trying to do some additional cleanup of unused aliases or if it's a foundation for IWYU analysis. If the former, I don't think IWYU should do it. If the latter, it makes sense to tackle separately, I think.

Reply:
Sure. This isn't a foundation for IWYU analysis.

Having played with the namespace reporting, I feel that as long as IWYU suggests a header to satisfy a reference to a namespace, then in order minimise unused headers IWYU also needs to identify unused aliases and using directives. I've got had some ideas on how we might be able to do this better.